modules/app_api/main.tf

```diff
@@ -0,0 +1,63 @@
+locals {
+  service_name   = "service-${var.name}"
+  container_name = "container-${var.name}"
+}
+
+data "aws_region" "current" {}
+
+resource "aws_cloudwatch_log_group" "default" {
+  #checkov:skip=CKV_AWS_158:Disabled to save money
+
+  name              = var.name
+  retention_in_days = 1
+}
+
+resource "aws_ecs_task_definition" "default" {
```

Comment:
IMO task definition should be defined in the application repository. It will be easier to trigger the deployment i.e. after we deploy a new image. It applies also if we choose another approach to deployment than terraform to reduce the risk of overwriting it from this repository.

 If we want to simplify it or define some defaults - maybe we should expose an `application` module that could be used by the app? A module can be fetched from [github](https://www.terraform.io/language/modules/sources#github) and it is possible to fetch from a [subdirectory](https://www.terraform.io/language/modules/sources#modules-in-package-sub-directories) so the `application` module definition could stay here.

*Possibly this could apply to the whole content of this module, as things such as exposed ports or the health check path are part of the application configuration.*